model/task_lifecycle.go

```diff
@@ -518,14 +518,29 @@ func MarkEnd(t *task.Task, caller string, finishTime time.Time, detail *apimodel
 		return errors.Wrap(err, "marking task finished")
 	}
 
-	if err = UpdateUnblockedDependencies(t, true, "MarkEnd"); err != nil {
-		return errors.Wrap(err, "updating unblocked dependencies")
+	unblockedIds := []string{}
+	if t.IsPartOfSingleHostTaskGroup() {
+		unblockedIds, err = UpdateUnblockedDependencies(t)
+		if err != nil {
+			return errors.Wrap(err, "updating unblocked dependencies")
+		}
 	}
 
-	if err = UpdateBlockedDependencies(t); err != nil {
+	blockedIds, err := UpdateBlockedDependencies(t)
+	if err != nil {
 		return errors.Wrap(err, "updating blocked dependencies")
 	}
 
+	unblockedAndNotBlockedIds, _ := utility.StringSliceSymmetricDifference(unblockedIds, blockedIds)
+	// The question to determine here is: do we ever unblock more dependencies than we block?
+	grip.DebugWhen(len(unblockedAndNotBlockedIds) > 0, message.Fields{
+		"message":                       "unblocked task group dependent tasks",
+		"ticket":                        "EVG-12923",
+		"unblocked_and_not_blocked_ids": unblockedIds,
```

Comment:
Is this meant to log `unblockedAndNotBlockedIds`, or all the unblocked ones (including the momentarily-unblocked ones)?

Reply:
Yes the former 🤦‍♀️